dist/Client.js

```diff
@@ -368,31 +368,110 @@ class APIClient {
    * @summary Creates a detector
    * @description Creates a detector setting its name and type
    * @param {String} name Name of the detector
-   * @param {String} detection_type Type of detection; on of "count", "segmentation"
+   * @param {String} detectionType Type of detection; one of "count", "segmentation"
+   * @param {String} outputType Type of results output; one of "bbox", "polygon"
+   * @param {String} trainingSteps Number of training steps, between 500 and 40000
    * @returns {Promise<String>} Id of the detector that has been created
    * @throws {APIError} Containing error code and text
    */
 
 
-  async createDetector(name = '', detection_type = 'count') {
+  async createDetector(name = '', detectionType = 'count', outputType = 'polygon', trainingSteps = 500) {
+    detectionType = detectionType.toLowerCase();
+    outputType = outputType.toLowerCase();
     const detectionTypes = ['count', 'segmentation'];
-    const type = detection_type.toLowerCase();
 
-    if (!detectionTypes.includes(type)) {
+    if (!detectionTypes.includes(detectionType)) {
       const validTypes = detectionTypes.join(', ');
-      throw new ValidationError(`Invalid detector type ${type}; allowed values: ${validTypes}.`);
+      throw new ValidationError(`Invalid detector type ${detectionType}; allowed values: ${validTypes}.`);
+    }
+
+    const outputTypes = ['polygon', 'bbox'];
+
+    if (!outputTypes.includes(outputType)) {
+      const validTypes = outputTypes.join(', ');
+      throw new ValidationError(`Invalid output type ${outputType}; allowed values: ${validTypes}.`);
+    }
+
+    const range = [500, 40 * 1000];
+
+    if (trainingSteps < range[0] || trainingSteps > range[1]) {
+      throw new ValidationError(`Training steps value ${trainingSteps} is outside range [${range[0]}, ${range[1]}].`);
     }
 
     const response = await this._request('/detectors/', 'POST', {
       'content-type': 'application/json'
     }, JSON.stringify({
       'name': name,
-      'detection_type': type
+      'detection_type': detectionType,
```

Comment:
Shouldn't this be under a `configuration` key ?

Reply:
Ah wrong branch...

Reply:
Pushed the good one